framework/include/utils/MooseTypes.h

```diff
@@ -934,3 +934,11 @@ DerivativeStringClass(ExtraElementIDName);
 
 /// Name of a Reporter Value, second argument to ReporterName (see Reporter.h)
 DerivativeStringClass(ReporterValueName);
+
+namespace Moose
+{
+extern const TagName SOLUTION_TAG;
+extern const TagName OLD_SOLUTION_TAG;
+extern const TagName OLDER_SOLUTION_TAG;
+extern const TagName PREVIOUS_NL_SOLUTION_TAG;
```

Comment:
Are all these new names necessary? I thought were trying to move away from the old, older, olderer, oldererer design

Reply:
What is the plan to move away from old, older, like stateful material props, old/older pps, etc.? The `Coupleable` interface can move away from `CoupledValueOld`, `implicit` etc. but it still requires the tag name to point which vector to be coupled in.

Reply:
Can't get by with a `TagID`? I don't know these systems very well so this may be a dumb question

Reply:
But I suppose even with a `TagID` that's just the type and not a value/identifier. Ok yea I'm sure this is fine

Reply:
These names can also be used by developers to provide the default tag name for calling `coupledVectorTagValue`. We can replace `implicit` parameter with `addParam<TagName>("tag", Moose::SOLUTION_TAG, "")`. Then we can replace `coupledValue("var")` with `coupledVectorTagValue("var", "tag")`. Thus we can keep only `coupledVectorTagValue` and users can use `tag = SOLUTION_1` to make a kernel implicit. The `tag` parameter can be a parameter for `Coupleable`, so we can simplify `coupledVectorTagValue("var", "tag")` back into `coupledValue("var")`. I think we are slowly moving to that direction. Does this make sense?